pkg/scalers/nsq_scaler.go

```diff
@@ -0,0 +1,358 @@
+	metadata   nsqMetadata
+	httpClient *http.Client
+	logger     logr.Logger
+}
+
+type nsqMetadata struct {
+	NSQLookupdHTTPAddresses  []string `keda:"name=nsqLookupdHTTPAddresses,  order=triggerMetadata;resolvedEnv"`
+	Topic                    string   `keda:"name=topic,                    order=triggerMetadata;resolvedEnv"`
+	Channel                  string   `keda:"name=channel,                  order=triggerMetadata;resolvedEnv"`
+	DepthThreshold           int64    `keda:"name=depthThreshold,           order=triggerMetadata;resolvedEnv, default=10"`
+	ActivationDepthThreshold int64    `keda:"name=activationDepthThreshold, order=triggerMetadata;resolvedEnv, default=0"`
+
+	triggerIndex int
+}
+
+const (
+	nsqMetricType = "External"
+)
+
+func NewNSQScaler(config *scalersconfig.ScalerConfig) (Scaler, error) {
+	metricType, err := GetMetricTargetType(config)
+	if err != nil {
+		return nil, fmt.Errorf("error getting scaler metric type: %w", err)
+	}
+
+	logger := InitializeLogger(config, "nsq_scaler")
+
+	nsqMetadata, err := parseNSQMetadata(config)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing NSQ metadata: %w", err)
+	}
+
+	return &nsqScaler{
+		metricType: metricType,
+		metadata:   nsqMetadata,
+		httpClient: kedautil.CreateHTTPClient(config.GlobalHTTPTimeout, true),
+		logger:     logger,
+	}, nil
+}
+
+func (m nsqMetadata) Validate() error {
+	if len(m.NSQLookupdHTTPAddresses) == 0 {
+		return fmt.Errorf("no nsqLookupdHTTPAddresses given")
+	}
+
+	if m.Topic == "" {
+		return fmt.Errorf("no topic given")
+	}
+
+	if m.Channel == "" {
+		return fmt.Errorf("no channel given")
+	}
+
+	if m.DepthThreshold <= 0 {
+		return fmt.Errorf("depthThreshold must be a positive integer")
+	}
+
+	if m.ActivationDepthThreshold < 0 {
+		return fmt.Errorf("activationDepthThreshold must be greater than or equal to 0")
+	}
+
+	return nil
+}
+
+func parseNSQMetadata(config *scalersconfig.ScalerConfig) (nsqMetadata, error) {
+	meta := nsqMetadata{triggerIndex: config.TriggerIndex}
+	if err := config.TypedConfig(&meta); err != nil {
+		return meta, fmt.Errorf("error parsing nsq metadata: %w", err)
+	}
+
+	return meta, nil
+}
+
+func (s nsqScaler) GetMetricsAndActivity(_ context.Context, metricName string) ([]external_metrics.ExternalMetricValue, bool, error) {
+	depth, err := s.getTopicChannelDepth()
+
+	if err != nil {
+		return []external_metrics.ExternalMetricValue{}, false, err
+	}
+
+	s.logger.Info("GetMetricsAndActivity", "metricName", metricName, "depth", depth)
+
+	metric := GenerateMetricInMili(metricName, float64(depth))
+
+	return []external_metrics.ExternalMetricValue{metric}, depth > s.metadata.ActivationDepthThreshold, nil
+}
+
+func (s nsqScaler) getTopicChannelDepth() (int64, error) {
+	nsqdHosts, err := s.getTopicProducers(s.metadata.Topic)
+	if err != nil {
+		return -1, fmt.Errorf("error getting nsqd hosts: %w", err)
+	}
+
+	if len(nsqdHosts) == 0 {
+		s.logger.Info("no nsqd hosts found for topic", "topic", s.metadata.Topic)
```

Comment:
This should be a debug info:
```suggestion
		s.logger.V(1).Info("no nsqd hosts found for topic", "topic", s.metadata.Topic)
```